### DISCUSSION TITLE
Safelist is ignored in JIT mode

### DISCUSSION THREAD

**Discussion by kolaente** [tailwindlabs/tailwindcss#5700] [Help] [answered]
Hi!

I'm using tailwind in JIT mode and it looks like it is ignoring the options i put in the safelist. My `tailwind.config.js` looks like this (removed irrelevant parts):

```js
module.exports = {
    mode: 'jit',
    theme: {
        extend: {
            listStyleType: {
                square: 'square',
                checkmark: '"✓"',
            },
        }
    },
    purge: {
        content: [
            '../../../../../../../vendor/hyva-themes/magento2-default-theme/**/*.phtml',
            '../../../../../../code/**/*.phtml',
            '../../../**/*.phtml',
        ],
        options: {
            safelist: [
                'list-checkmark',
                'marker:text-brand-primary',
            ],
        },
    }
}
```

The problem with the options in the safelist is they are only used in cms content coming from a database, so tailwind has no way of finding them for purging. I could just put them in a file somewhere and add that to the purging options but that feels a lot like a hack.

Looks similar to https://github.com/tailwindlabs/tailwindcss/issues/2829#issuecomment-897568571, not sure if that's a bug.

**Comment by simonswiss** [comment, marked as answer]
Hey!

While this _should_ work, I was able to reproduce your issue locally. With JIT, we have also added the ability to have the `safelist` directly in the `purge` object, instead of in `purge.options`.

The `options` was a way to pass raw options to PurgeCSS directly, and JIT doesn't use PurgeCSS anymore, hence the change.

This will change again for Tailwind 3.0 (where JIT will be the _default_ and _only_ mode), but for now, you can fix your config by moving your `safelist` array one level up, directly in the `purge` object:

```diff
purge: {
+  safelist: ['text-red-500'],
-  options: {
-    safelist: ['text-red-500']
-  }
}
```

Hope it helps!

**Reply by kolaente** [reply]
That works. Thanks a lot!

**Reply by jlsrvr** [reply]
Hello, thanks for the helpful answer!
You mention that this will change for Tailwind 3.0 but it seems to be working at the moment in v 3.0.13. 
Is there any plan to change the way this works?
